service/spotify_service.go

```
@@ -0,0 +1,164 @@
+		return spotiferr.ResponseError
+	}
+
+	//Unmarshall JSON is use to MAP the values from body json to the Struct ArtistResponseJSON
+	jsonResponse := ArtistResponseJSON{}
+	if err := json.Unmarshal([]byte(string(body)), &jsonResponse); err != nil {
+		return spotiferr.UnmarshallError
+
+	}
+
+	// Search through ITEM array (from Response body) and find the Band/artist Name.
+	for _, itemArtist := range jsonResponse.Artists.Items {
+		if strings.ToLower(itemArtist.Name) == strings.ToLower(artist) {
+			targetArtist.ID = itemArtist.SpotifyID
+			targetArtist.Name = itemArtist.Name
+			targetArtist.SpotifyURL = itemArtist.ExternalUrls.Spotify
+
+			//  Genres is a []string
+			// []string... Let me pass multiple values to the func append.
+			targetArtist.Genres = append(targetArtist.Genres, itemArtist.Genres...)
+		}
+	}
+
+	if targetArtist.Name == "" {
+		return spotiferr.ArtistNotFound
+	}
+
+	return nil
+
+}
+
+// RepreshResponseJSON is a struct that handle the "unmarshall" information from HTTP Response Body
+// of RefreshToken.
+type RefreshTokenResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+}
+
+// RefreshToken: Make a HTTP POST call to  Spotify API to get a new Token.
+// this func will return an string (refresh token) and an error.
+func RefreshToken() (string, error) {
+	//Data for the HTTP POST CALL
+	postData := url.Values{
+		"grant_type":    {"refresh_token"},
+		"refresh_token": {SpotifyRefreshToken},
+	}
+
+	//Create Request HTTP POST CALL structure with data and headers.
+	request, err := http.NewRequest("POST", SpotifyRefreshEndpoint, strings.NewReader(postData.Encode()))
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	request.Header.Set("Authorization", SpotifyAuhtorizationToken)
+	if err != nil {
+		return "", spotiferr.BadRequestFormat
+	}
+
+	//HTTP Client makes the request to the Spotify API Endpoint.
+	response, _ := http.DefaultClient.Do(request)
```

Never ignore errors.